pokedex/storage.lua

```diff
@@ -61,49 +98,41 @@ function M.clamp_max_active_pokemon(new_max)
 	return math.max(range_min, math.max(range_min, new_max))
 end
 
+
 function M.is_party_full()
 	local counter = 0
-	for _, _ in pairs(active) do
-		counter = counter + 1
+	for _, pkmn in pairs(player_pokemon) do
+		if pkmn.location == LOCATION_PART then
+			counter = counter + 1
+		end
 	end
 	return counter >= M.get_max_active_pokemon()
 end
 
-local function sort_on_index(a, b)
-	return function(a, b) 
-		local c = pokedex.get_index_number(a.species.current)
-		local d = pokedex.get_index_number(b.species.current)
-		return c < d  
-	end
-end
-
-local function sort_on_caught(a, b)
-	return function(a, b) 
-		return a.number < b.number  
-	end
-end
-
-local function sort_on_level(a, b)
-	return function(a, b) return a.level.current > b.level.current end
-end
-
-local function sort_alphabetical(a, b)
-	return function(a, b) return a.species.current < b.species.current end
-end
 
 function M.list_of_ids_in_storage()
 	local f = M.get_sorting_method()
-	return getKeysSortedByValue(storage, f(a, b))
+	local tmp = {}
+	for id, pkmn in pairs(player_pokemon) do
+		if pkmn.location == LOCATION_PC then
+			tmp[id] = pkmn
+		end
+	end
+	return getKeysSortedByValue(tmp, f(a, b))
 end
 
-local function sort_on_slot(a, b)
-	return function(a, b) return (a.slot or 7) < (b.slot or 7) end
-end
 
 function M.list_of_ids_in_inventory()
-	return getKeysSortedByValue(active, sort_on_slot(a, b))
+	local tmp = {}
+	for id, pkmn in pairs(player_pokemon) do
```

Comment:
Should use the individual location tables instead, see above comments. This one is especially egregious, as checking on your party should NOT hit all 10,000 pokemon in your PC.